Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/ISISIndirectEnergyTransferWrapperTest.py

```diff
@@ -0,0 +1,313 @@
+                                                      Instrument='IRIS',
+                                                      Analyser='graphite',
+                                                      Reflection='002',
+                                                      SpectraRange=[3, 53])
+
+        workspace = ISISIndirectEnergyTransferWrapper(InputFiles=['IRS26176.RAW'],
+                                                      Instrument='IRIS',
+                                                      Analyser='graphite',
+                                                      Reflection='002',
+                                                      SpectraRange=[3, 53],
+                                                      Efixed=1.845)
+
+        self.assertTrue(CompareWorkspaces(reference, workspace)[0])
+
+    def test_that_a_reduction_with_scale_factor_will_produce_a_workspace_with_the_correct_number_of_histograms(self):
+        workspace = ISISIndirectEnergyTransferWrapper(InputFiles=['IRS26176.RAW'],
+                                                      Instrument='IRIS',
+                                                      Analyser='graphite',
+                                                      Reflection='002',
+                                                      SpectraRange=[3, 53],
+                                                      ScaleFactor=0.5)
+
+        red_workspace = workspace.getItem(0)
+        self.assertEqual(red_workspace.getNumberHistograms(), 51)
+
+    def test_that_the_history_contains_only_one_parent_algorithm_which_has_the_correct_name(self):
+        workspace = ISISIndirectEnergyTransferWrapper(InputFiles=['IRS26176.RAW'],
+                                                      Instrument='IRIS',
+                                                      Analyser='graphite',
+                                                      Reflection='002',
+                                                      SpectraRange=[3, 53])
+
+        history = workspace.getItem(0).getHistory()
+        algorithm_names = [alg.name() for alg in history]
+
+        self.assertEqual(len(algorithm_names), 1)
+        self.assertEqual(algorithm_names[0], 'ISISIndirectEnergyTransferWrapper')
+
+    def test_that_the_history_of_ISISIndirectEnergyTransferWrapper_has_no_child_algorithms(self):
+        workspace = ISISIndirectEnergyTransferWrapper(InputFiles=['IRS26176.RAW'],
+                                                      Instrument='IRIS',
+                                                      Analyser='graphite',
+                                                      Reflection='002',
+                                                      SpectraRange=[3, 53])
+
+        history = workspace.getItem(0).getHistory()
+
+        child_histories = history.getAlgorithmHistory(0).getChildHistories()
+        algorithm_names = [alg.name() for alg in child_histories]
+
+        self.assertFalse(algorithm_names)
```

Comment:
I think we should check some of the values of the workspaces in the tests, to make sure it does produce the correct reduced data